frontends/ol-search-ui/src/components/Searchbox.tsx

```diff
@@ -0,0 +1,191 @@
+import { ValidationError } from "ol-forms"
+import React, { useCallback, useState } from "react"
+import styled, { css } from "styled-components"
+
+const SearchboxContainer = styled.div`
+  display: flex;
```

Comment:
I don't think we want the styles in the component  code. Can this be moved to an scss file? Mixing css in javascript files isn't great

Reply:
Re ... Some of these comments from slack (Though I reordered some of these.)

> 1. Is the first place we’re using Styled Components right?
> 2. Mixing css in javascript files isn't great
> 3. I don’t think it’s a good idea generally, since it makes it harder to re-use css code among components
> 4. It's very likely that the styling we eventually have for the infinite corridor site will be different than open discussions
> 5. And also having styles in open-discussions work differently from all our other codebases is undesirable

**Re 1:** This isn't the first place we are using styled-components, though looking back, our usage is much more limited than I realized. We were originally using it in PageBanner.js (which I converted to tsx) and AuthEmailForm.js. We also use it (very limitedly) in mitxonline.

**Re 2-4:** Re mixing css and javascript... One thing that I've heard people say about this in the past is that this is mixing concerns, and we want separations of concerns.

In fact, I think this sort of pattern is a much better separation of concerns: We co-locate the styling specific to Searchbox with the javascript code specific to Searchbox. (Whereas previously the styling and layout/js were tightly coupled but in very different areas of our codebase.) Before, we were separating technologies. Now, we are isolating (and separating) concerns: All the searchbox code is together. 

I don't mean to suggest I've done this perfectly, by the way. E.g., the `height` properties in `SearchboxContainer` is a little awkward and that's probably better handled some other way, but I'd rather cross that bridge when we get to it. To a large extent, this the css is searchbox specific and makes sense to have with the searchbox component. (Example: the alignment of the icons within the textbox.¹) 

Re sharing styles and different styles for open-discussions / infinite-corridor: We can definitely do this with styled components via themes. I set it up in this PR... e.g., see [color based on theme here](https://github.com/mitodl/open-discussions/pull/3614/files#diff-4fd28178e751ae5b887c461130e5b555a8654a9b22d29e84e13104fb92148697R39)

**Re 5**: Yeah, that's definitely a concern. As mentioned, I just saw we had styled-components install previously, I did not realize our usage was so limited.

**All in all**: I definitely think we should be isolating component-specific and co-locatting all the relevant code for a single component. Styled components is a pretty easy way to do this. There are other solutions, e.g., css / scss modules.

------

¹I mean, the alignment of icons in the searchbox could certainly be abstracted to a different component... InputWithIcons. But that's more refactoring than I think we should do at the moment. Or better yet we could find a library for that.

Reply:
@abeglova I just put up one more commit that moves some of the course searchbox styling back to open-discussions. 

I've tried to keep the stuff that really is searchbox specific (example: css related to positioning the icons) isolated / co-located with the component. But now you can apply a className the to component to set stuff like font size, border radius. Some things maybe could still be moved out (e.g., the 500px max-width) but imo we should defer that movement till its needed.